File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/SendContractWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.SendContractUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

@Component
public class SendContractWorker {

    private final SendContractUseCase useCase;

    public SendContractWorker(SendContractUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.SEND_CONTRACT)
    public void sendContract(@Variable String applicationId) {
        useCase.sendContract(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/SendContractWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.SendContractUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

@Component
class SendContractWorker(
    private val useCase: SendContractUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.SEND_CONTRACT)
    fun sendContract(@Variable applicationId: String) {
        useCase.sendContract(ApplicationId.of(applicationId))
    }
}
